Automattic/jetpack
# Refactor/memberships/user is paid subscriber

Fixes 107-gh-automattic/gold
Blocks D137961-code

## Proposed changes:

* Pass an optional `$user_id` to `subscription_service()`
* Use the `WPCOM_Online_Subscription_Service` when `$user_id` is provided, even if it is not `available`.
* Store this `$user_id` in the `WPCOM_Online_Subscription_Service` object.
* Use this `$user_id` to determine if the user can view content if the user is not logged in.

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Apply D137961-code to your sandbox 
* Download this patch on top of D137961-code
* Visit a newsletter post that is set to some tier
* Purchase this tier as a subscriber
* Open a new private window and return to the post
* Enter the subscriber's email address and click "upgrade"
* You should be prompted for a login code and you should be able to login after entering that code.
